### PR TITLE
Add braces in switch sections

### PR DESCRIPTION
**Pull request by thomaslevesque**
#252

### CODE REVIEW

File: src/CSharp/CodeCracker/Refactoring/AddBracesToSwitchSectionsAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace CodeCracker.Refactoring
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class AddBracesToSwitchSectionsAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "CC0073";
```
Comment (giggio) [inline src/CSharp/CodeCracker/Refactoring/AddBracesToSwitchSectionsAnalyzer.cs]: Needs to use the helper method. Check out the other analyzers.

File: src/CSharp/CodeCracker/Refactoring/AddBracesToSwitchSectionsAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            DiagnosticSeverity.Hidden,
+            isEnabledByDefault: true,
+            helpLink: HelpLink.ForDiagnostic(CodeCracker.DiagnosticId.AddBracesToSwitchSections));
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(Rule); } }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.SwitchStatement);
+        }
```
Comment (giggio) [inline src/CSharp/CodeCracker/Refactoring/AddBracesToSwitchSectionsAnalyzer.cs]: Use the expression syntax. See other analyzers.

File: test/CSharp/CodeCracker.Test/CodeCracker.Test.csproj [hunk 3]
```diff
@@ -213,9 +214,6 @@
     <Folder Include="Portability\" />
     <Folder Include="Security\" />
   </ItemGroup>
-  <ItemGroup>
-    <Analyzer Include="..\..\..\packages\codecracker.CSharp.1.0.0-alpha3-0346\tools\analyzers\CodeCracker.CSharp.dll" />
-  </ItemGroup>
```
Comment (giggio) [inline test/CSharp/CodeCracker.Test/CodeCracker.Test.csproj]: Don't remove the analyzer.

**Comment by giggio** [comment]
Some tests are failing: https://ci.appveyor.com/project/code-cracker/code-cracker/build/1.0.0.364/tests

File: test/CSharp/CodeCracker.Test/CodeCracker.Test.csproj (hunk 3 above)
Comment (thomaslevesque) [inline test/CSharp/CodeCracker.Test/CodeCracker.Test.csproj]: Oops, sorry. I didn't intend to commit that change...

**Comment by thomaslevesque** [comment]
Oops, forgot to re-run the tests after last minute changes to the message...

**Comment by thomaslevesque** [comment]
All fixed

**Comment by thomaslevesque** [comment]
Don't merge it yet, I'll rebase it to fix the conflicts myself

**Comment by thomaslevesque** [comment]
Done, ready to merge

**Pull request merged**